odoo/http.py

```diff
@@ -915,6 +914,25 @@ def first_arg_is_req(self):
     def __call__(self, *args, **kw):
         return self.method(*args, **kw)
 
+    # werkzeug will use these EndPoint objects as keys of its RoutingMap._rules_by_endpoint mapping.
+    # When Odoo clears the routing map, new EndPoint objects are created, most of them with the same values.
+    # The __eq__ and __hash__ magic methods allow older EndPoint objects to be still valid keys of the RoutingMap.
+    # For example, website._get_canonical_url_localized may use such an old endpoint if the routing map was cleared.
```

Comment:
Nitpicking for next time: Let's try to not go over 80 char in docstring and comments